tb_plugin/torch_tb_profiler/profiler/module_op.py

```diff
@@ -0,0 +1,220 @@
+from copy import copy
+
+from .node import ModuleNode, ProfilerStepNode
+from .trace import EventTypes
+
+
+class Module:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
+        self.children = []
+
+    def __hash__(self):
+        if self.children:
+            return hash((self.name, self.module_id, tuple(self.children)))
+        else:
+            return hash((self.name, self.module_id))
+
+    def __eq__(self, o) -> bool:
+        if not isinstance(o, Module):
+            return False
+
+        return (self.name == o.name and
+                self.module_id == o.module_id and
+                self.children == o.children)
+
+class ModuleStats:
+    def __init__(self, name, module_id):
+        self.name = name
+        self.module_id = module_id
```

Comment:
`module_id` should be distinct so not sure how you can have multiple `occurences`? i.e. you'll see one `Linear_0` and not like 3 `Linear_0` s

Reply:
the occurences happens across multiple steps. For example, the Sequential_0 repeat 10 times.